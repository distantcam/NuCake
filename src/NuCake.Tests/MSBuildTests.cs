using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

[UseReporter(typeof(DiffReporter))]
public class MSBuildTests
{
    [Fact]
    public void NoAttributesGivesWarnings()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj"));
    }

    [Fact]
    public void SettingTitleRemovesWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetTitle"));
    }

    [Fact]
    public void SettingDescriptionRemovesWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetDescription"));
    }

    [Fact]
    public void SettingCompanyRemovesWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetCompany"));
    }

    [Fact]
    public void SettingEverythingRemovesAllWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetTitle", "SetDescription", "SetCompany", "SetInformationalVersion"));
    }
}