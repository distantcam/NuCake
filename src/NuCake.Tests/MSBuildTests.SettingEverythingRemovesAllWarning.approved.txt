Project "[PROJECT_DIRECTORY]\SampleProject.csproj" on node 1 (Rebuild target(s)).
CoreClean:
  Deleting file "[PROJECT_DIRECTORY]\bin\Debug\SampleProject.dll".
  Deleting file "[PROJECT_DIRECTORY]\bin\Debug\SampleProject.pdb".
  Deleting file "[PROJECT_DIRECTORY]\obj\Debug\SampleProject.dll".
  Deleting file "[PROJECT_DIRECTORY]\obj\Debug\SampleProject.pdb".
  Deleting file "[PROJECT_DIRECTORY]\obj\Debug\SampleProject.csprojResolveAssemblyReference.cache".
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
  C:\Windows\Microsoft.NET\Framework\v4.0.30319\Csc.exe /noconfig /nowarn:1701,1702 /nostdlib+ /platform:AnyCPU /errorreport:prompt /warn:4 /define:SetTitle;SetDescription;SetCompany;SetInformationalVersion /highentropyva- /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\mscorlib.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.dll" /debug+ /debug:full /filealign:512 /optimize- /out:obj\Debug\SampleProject.dll /target:library /utf8output AssemblyInfo.cs "C:\Users\cameron.macfarland\AppData\Local\Temp\.NETFramework,Version=v4.0.AssemblyAttributes.cs"
CopyFilesToOutputDirectory:
  Copying file from "obj\Debug\SampleProject.dll" to "bin\Debug\SampleProject.dll".
  SampleProject -> [PROJECT_DIRECTORY]\bin\Debug\SampleProject.dll
  Copying file from "obj\Debug\SampleProject.pdb" to "bin\Debug\SampleProject.pdb".
CreateNuGetPackage:
  Package created -> [PROJECT_DIRECTORY]\NuGetBuild\SampleProject 1.0.0-info.nupkg
Done Building Project "[PROJECT_DIRECTORY]\SampleProject.csproj" (Rebuild target(s)).
Build succeeded.
    0 Warning(s)
    0 Error(s)